scripts/asm-installer/install_asm

```diff
@@ -2028,6 +2025,12 @@ generate_membership_name() {
 register_cluster() {
   if is_cluster_registered; then return; fi
 
+  if can_modify_gcp_components; then
```

Comment:
Can we instead just move register cluster after WI enablement. So that we don't have this duplicate logic here.

Reply:
It would need to go at multiple places in main because the path splits based off of various things, I'd rather have it here just in case to prevent something like this from happening again.

Reply:
Just two branches right? One under `is_managed`, the other one `! is_managed`.

Reply:
For now. I don't know about you, but I'm certainly not confident that I'm going to be able to keep track of the next time a branch changes in 2.5k lines of bash. I'd rather not worry about it for the same of a few redundant lines.

Reply:
fair enough...